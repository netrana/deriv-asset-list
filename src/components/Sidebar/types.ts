type MenuItem = {
  title: string;
  icon: JSX.Element;
  link: string;
  className?: string;
  parentRoute?: string;
};

export interface Props {
  menuItems: MenuItem[];
  className?: string;
}
