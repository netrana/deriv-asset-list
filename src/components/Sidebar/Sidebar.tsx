import React, { FC, useEffect } from 'react';
import { Layout, Menu } from 'antd';
import { Link, useLocation } from 'react-router-dom';
import classNames from 'classnames';

import styles from './sidebar.module.scss';

import { Props } from './types';

const { Sider } = Layout;

const Sidebar: FC<Props> = (props) => {
  const { menuItems, className = '' } = props;

  const location = useLocation();

  const [selectedTab, setSelectedTab] = React.useState('0');

  useEffect(() => {
    setSelectedTab(
      menuItems
        .findIndex((item) =>
          location.pathname.includes(!!item.parentRoute ? item.parentRoute : item.link),
        )
        .toString(),
    );
  }, [location.pathname]);

  return (
    <Sider   
      width={210}   
      breakpoint="xs"
      collapsedWidth="0"
      className={classNames({
        [styles.sidebarContainer]: true,
        [className]: Boolean(className),
      })}
    >
      <Menu className={styles.sidebarMenus} mode="inline" selectedKeys={[selectedTab]}>
        {menuItems.map((menuItem, index) => (
          <Menu.Item key={index} icon={menuItem.icon}>
            {menuItem.title}
            <Link to={menuItem.link} />
          </Menu.Item>
        ))}
      </Menu>
    </Sider>
  );
};

export default Sidebar;
