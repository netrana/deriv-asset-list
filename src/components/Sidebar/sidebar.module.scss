@import "styles/globals.scss";

.sidebarContainer {
  background-color: #fafafa;
  font-family: "Poppins", sans-serif;
  font-style: normal;
  font-weight: normal;
  height: 100%;
  border-right: 1px solid #d9d9d9;
  bottom: 0;
  width: 220px;
  padding-top: 61px;

  :global(.ant-menu-item-selected) {
    background-image: $backgroundImage !important;
    color: white;
  }

  :global(.ant-menu-item) {
    height: 40px;
    line-height: 22px;
    font-size: 16px;
    margin:0 !important;
  }

  :global(.ant-menu-item:hover) {
    background-image: $backgroundImage !important;
    color: white;
  }

  :global(.ant-menu-item::after){
    border-right: 0ch;
  }
}

.sidebarMenus {
  background-color: #fafafa;
}
