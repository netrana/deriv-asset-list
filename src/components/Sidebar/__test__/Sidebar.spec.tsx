import React from "react";
import render from "react-test-renderer";

import Sidebar from "../Sidebar";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  FC: jest.fn(),
  useEffect: jest.fn(),
}));

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "/",
  }),
  Link: "Link",
}));

jest.mock("antd", () => ({
  ...jest.requireActual("antd"),
  Menu: "Menu",
  Layout: {
    Sider: "Sider",
  },
}));

describe("Sidebar component test", () => {
  describe("snapshot", () => {
    it("should match snapshot", () => {
      const tree = render.create(<Sidebar menuItems={[]} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
