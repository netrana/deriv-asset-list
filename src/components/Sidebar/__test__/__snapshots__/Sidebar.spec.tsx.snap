// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component test snapshot should match snapshot 1`] = `
<Sider
  breakpoint="xs"
  className="sidebarContainer"
  collapsedWidth="0"
  width={210}
>
  <Menu
    className="sidebarMenus"
    mode="inline"
    selectedKeys={
      Array [
        "0",
      ]
    }
  />
</Sider>
`;
